test/Silex/ApiProblemProviderTest.php

```diff
@@ -41,23 +44,76 @@ public function serviceProvider() : Traversable
 
     /**
      * @test
+     * @dataProvider exceptionProvider
      */
-    public function it_handles_errors()
+    public function it_handles_http_exceptions(bool $includeExceptionDetails)
     {
+        if (!$includeExceptionDetails) {
+            $this->createApplication(false);
+        }
+
         $client = $this->createClient();
 
         $client->request('GET', '/error');
         $response = $client->getResponse();
 
-        $expected = [
-            'type' => 'about:blank',
-            'title' => 'problem',
-        ];
+        if ($includeExceptionDetails) {
+            $expected = [
+                'exception' => 'problem',
+                'stacktrace' => $this->exception->getTraceAsString(),
```

Comment:
being this a JSON dictionary, could a structured version like `getTrace()` be used or is it ugly?
Also, I'm not sure the trace includes the information from `getLine()` and `getFile()`.

Reply:
And since we are in a debug mode we can possibly include also `getMessage()` in the response rather than just in the log? (If I understood correctly)

Reply:
👍